crash in subnet matrix

https://uninav.uninett.no/report/matrix?scope=128.39.57.0/24
https://uninav.uninett.no/report/matrix?scope=128.39.57.0/24&show_unused_addresses=True

AssertionError: 128.39.57.0/24 <= than 24

Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 111, in get_response response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/usr/lib/python2.7/dist-packages/nav/web/report/views.py", line 171, in matrix_report matrix = create_matrix(scope, show_unused)
File "/usr/lib/python2.7/dist-packages/nav/web/report/views.py", line 243, in create_matrix matrix.build()
File "/usr/lib/python2.7/dist-packages/nav/report/matrixIPv4.py", line 44, in build for net in nets
File "/usr/lib/python2.7/dist-packages/nav/report/matrixIPv4.py", line 62, in _write_subnets subnets = IPtools.create_subnet_range(net, self._get_row_size())
File "/usr/lib/python2.7/dist-packages/nav/report/IPtools.py", line 296, in create_subnet_range assert prefixlen > net.prefixlen(), '{} <= than {}'.format(net, prefixlen)
AssertionError: 128.39.57.0/24 <= than 24

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2017-01-12T12:45:27Z
 * owner: vegard-vesterheim
 * the launchpad url was https://bugs.launchpad.net/bugs/1655949

--- a/python/nav/report/IPtools.py
+++ b/python/nav/report/IPtools.py
@@ -293,11 +293,13 @@ def get_next_subnet(net):
 
 def create_subnet_range(net, prefixlen):
     """Creates all subnets of the given size inside the net"""
-    assert prefixlen > net.prefixlen(), '{} <= than {}'.format(net, prefixlen)
+    assert prefixlen >= net.prefixlen(), '{} < than {}'.format(net, prefixlen)
+    # Return self as the net cannot be divided further
+    if net.prefixlen() == prefixlen:
+        return [net]
     subnet = IP("{}/{}" .format(net.net().strNormal(), prefixlen))
     subnet_range = []
     while net.overlaps(subnet):
         subnet_range.append(subnet)
         subnet = get_next_subnet(subnet)
-
     return subnet_range

--- a/python/nav/report/matrixIPv4.py
+++ b/python/nav/report/matrixIPv4.py
@@ -59,9 +59,10 @@ class MatrixIPv4(Matrix):
 
         # Initially, create the rows (subnets) we're going to display
         if self.show_unused_addresses:
-            subnets = IPtools.create_subnet_range(net, self._get_row_size())
+            row_size = self._get_row_size()
+            subnets = IPtools.create_subnet_range(net, row_size)
             large_subnets = [x for x in nets.keys()
-                             if x.prefixlen() < self._get_row_size()]
+                             if x.prefixlen() < row_size]
         else:
             subnets = IPtools.sort_nets_by_address(nets.keys())
 

